### ISSUE TITLE
Enable CCACHE for AppVeyor Windows builds

### ISSUE THREAD

**Issue by larsbergstrom** [servo/servo#9874]
Appveyor now builds Servo. See: https://ci.appveyor.com/project/servo/servo/build/1.0.17

We could speed it up significantly by caching the SpiderMonkey build artifacts. Here, we'd set the `CCACHE` environment variable, as done similarly for Travis here:
https://github.com/larsbergstrom/servo/commit/5c5acc02498ebd14003f086fdb38f56bd5c4e070

And then add it to the list of cached directories between builds here:
https://github.com/servo/servo/blob/master/appveyor.yml#L10

Note that this should be tested first on a Windows installation (virtual machine or native should both be fine), and you should use `ccache -s` after a build against the cache to ensure that there are a substantial number of `cache hit` successes before opening a PR.

**Comment by sorpaas** [comment]
I would like to work on this. @larsbergstrom

**Comment by larsbergstrom** [comment]
@sorpaas That sounds great - thanks! Please let us know if you have any problems with it. I know that the MSYS/MINGW environment can be pretty tricky.

**Comment by jdm** [comment]
@sorpaas Any progress?

**Comment by sorpaas** [comment]
Really sorry @jdm. I'm suffering from limited and unstable Internet access until next month. I really wouldn't mind it if someone take over this issue.

**Comment by jdm** [comment]
Ok, thanks for letting us know!

**Comment by amarmaduke** [comment]
I'd like to work on this. @larsbergstrom

**Comment by jdm** [comment]
Our Windows builds are taking a long time right now; have you made any progress @amarmaduke? Any questions?

**Comment by amarmaduke** [comment]
I'm running into trouble with the mingw64 build @jdm

The build fails on the build command for `mozjs_sys v0.0.0 (https://github.                                                    com/servo/mozjs#5ac99916)`

It says:

```
Build failed, waiting for other jobs to finish...
error: failed to run custom build command for `mozjs_sys v0.0.0 (https://github.com/servo/mozjs#5ac99916)`
Process didn't exit successfully: `C:/msys64/home/Andrew/servo/target\debug\build\mozjs_sys-736cac28b49f9dab\build-script-build` (exit code: 101)
```

Part of the stdout (I left out bits because of length):

```
cd /home/Andrew/servo/target/debug/build/mozjs_sys-736cac28b49f9dab/out && \
PYTHON=""c:/python27/python.exe"" \
MOZ_TOOLS="/" CC="gcc" CPP="gcc -E" CXX="g++" AR="ar" \
/home/Andrew/servo/.cargo/git/checkouts/mozjs-06d7f04b6dbb8a8e/master/mozjs/js/src/configure --disable-jemalloc --disable-shared-js --disable-export-js
creating cache ./config.cache
checking host system type... x86_64-pc-mingw64
.
.
.
checking for setlocale... yes
checking for localeconv... (cached) yes
checking for malloc_size... no
checking for malloc_usable_size... no
updating cache ./config.cache
creating ./config.status
makefile.cargo:75: recipe for target 'all' failed
```

The stderr is:

```
makefile.cargo:58: extraneous text after 'else' directive
/home/Andrew/servo/.cargo/git/checkouts/mozjs-06d7f04b6dbb8a8e/master/mozjs/js/src/configure: line 3772: test: (GNU: integer expression expected
Traceback (most recent call last):
  File "C:/msys64/home/Andrew/servo/.cargo/git/checkouts/mozjs-06d7f04b6dbb8a8e/master/mozjs/build/subconfigure.py", line 422, in <module>
    sys.exit(main(sys.argv[1:]))
  File "C:/msys64/home/Andrew/servo/.cargo/git/checkouts/mozjs-06d7f04b6dbb8a8e/master/mozjs/build/subconfigure.py", line 406, in main
    return subconfigure(args)
  File "C:/msys64/home/Andrew/servo/.cargo/git/checkouts/mozjs-06d7f04b6dbb8a8e/master/mozjs/build/subconfigure.py", line 393, in subconfigure
    pool.imap_unordered(run, subconfigures):
  File "C:\python27\Lib\multiprocessing\pool.py", line 668, in next
    raise value
IOError: [Errno 2] No such file or directory: 'intl/icu/target/common/Makefile'
make: *** [all] Error 1
thread '<main>' panicked at 'assertion failed: result.success()', C:/msys64/home/Andrew/servo/.cargo\git\checkouts\mozjs-06d7f04b6dbb8a8e\master\build.rs:17
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

I'll have time to investigate it more thoroughly this Sunday, but perhaps you know some common mistakes that this is the symptom of?

**Comment by larsbergstrom** [comment]
@amarmaduke Is this a from-scratch Windows build, or have you started working on the CCACHE parts? 

I have talked with our Windows Gecko contributors and found out that CCACHE support on Windows is likely not to work due to some vagarities of the build system. So, unfortunately, I will need to close this task! I'm really sorry - I had quickly investigated and it all seemed to be straightforward, but I discovered that there are some uses of python and path-related issues that will prevent it from being a non-trivial task (if it is possible at all).
